autoload/go/def.vim

```diff
@@ -114,11 +135,122 @@ function! s:godefJump(out, mode)
 			endif
 		endif
 
-		" jump to file now
-		sil ll 1
-		normal zz
+        " Remove anything newer than the current position, just like basic
+        " vim tag support
+        if w:go_stack_level == 0
+            let w:go_stack = []
+        else
+            let w:go_stack = w:go_stack[0:w:go_stack_level-1]
+        endif
+
+		" increment the stack counter
+		let w:go_stack_level = w:go_stack_level + 1
```

Comment:
Fixed the go_stack_level.  Would I be okay to do a :retab as a final step to fix the indentation?  What's the format you're using?